src/Build/Evaluation/Expander.cs

```diff
@@ -799,10 +799,6 @@ internal static string ExpandMetadataLeaveEscaped(string expression, IMetadataTa
 
                     return result;
                 }
-                catch (ArgumentException ex) // Blank metadata name
```

Comment:
@rainersigwald 
I decided to not catch `ArgumentException` here, since I could not get it failing with blank metadata. Also, this method does a bit more than the other try catch protected places which just compute metadata.